### PR TITLE
FIX Add classes_ attribute to AdversarialFairnessClassifier

### PR DESCRIPTION
**Pull request by TamaraAtanasoska**
## Description
This PR addresses issue #1381.
Additionally, the new scikit-learn estimator checks are enabled for both `AdversarialFairnessClassifier` and `AdversarialFairnessRegressor`. As a result many checks are failing and they are added under the `xfail` flag. They would be addressed individually in the future.

@adrinjalali 

## Tests

- [ ] no new tests required
- [x] new tests added
- [x] existing tests adjusted

## Documentation

- [ ] no documentation changes needed
- [ ] user guide added or updated
- [x] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

**Review by romanlutz — APPROVED** [review]

**Comment by TamaraAtanasoska** [comment]
@adrinjalali CI is green-ish, openml failed right at the end. Would you mind retriggering just those two last jobs? 

If you don't mind I will do a few test running CI in PR #1385 to check what is the state of our code and the new numpy API compatibility, I'll make new branch with fixes if there are issues.

File: fairlearn/adversarial/_adversarial_mitigation.py [hunk 1]
```diff
@@ -540,7 +541,7 @@ def fit(self, X, y, *, sensitive_features=None):
 
         return self
 
-    def partial_fit(self, X, y, *, sensitive_features=None):
+    def partial_fit(self, X, y, *, sensitive_features=None, classes=None):
```
Comment (adrinjalali) [inline fairlearn/adversarial/_adversarial_mitigation.py:544]: where is this then used?

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-numpy>=1.24.4
+numpy~=1.24.4
```
Comment (adrinjalali) [inline requirements.txt:1]: O_o, why not less than 2?

File: fairlearn/adversarial/_adversarial_mitigation.py (hunk 1 above)
Comment (TamaraAtanasoska) [reply fairlearn/adversarial/_adversarial_mitigation.py:544]: The estimator checks fail if it isn't there, and I found it the same way in scikit-learn's code. I assumed it must be used somewhere, I am not very familiar with `partial fit `and it use cases 😬

File: requirements.txt (hunk 2 above)
Comment (TamaraAtanasoska) [reply requirements.txt:1]: For some reason, CI would still fail with [066218e](https://github.com/fairlearn/fairlearn/pull/1384/commits/066218e7c9fe9bf1138db6f450d8c01277a7517f). This is how I realised we have `numpy` API incompatibilities in the tests. Changing it to [937e09c](https://github.com/fairlearn/fairlearn/pull/1384/commits/937e09c6ff9f39a48727d0afe70a624ae7d5af77) makes CI pass. I was also confused why this is the case, but as I am already working on figuring this out on a more deeper level I though it won't hurt temporarily :)

File: fairlearn/adversarial/_adversarial_mitigation.py (hunk 1 above)
Comment (TamaraAtanasoska) [reply fairlearn/adversarial/_adversarial_mitigation.py:544]: scikit-learn example: https://github.com/scikit-learn/scikit-learn/blob/329a1cf4bd46664fbf81806cde82a27f44b760e0/sklearn/multiclass.py#L394C32-L394C33

File: requirements.txt (hunk 2 above)
Comment (TamaraAtanasoska) [reply requirements.txt]: I reverted to range [4cbaf64](https://github.com/fairlearn/fairlearn/pull/1384/commits/4cbaf6458c873a9efc302a1a74132d9a8bc9a53e), which should fail only the `tensorflow` `build-test/other-ml` build. That one only seems to pass if we keep the version low as here `numpy~=1.24.4`.

File: fairlearn/adversarial/_adversarial_mitigation.py (hunk 1 above)
Comment (adrinjalali) [reply fairlearn/adversarial/_adversarial_mitigation.py:544]: Yes I know they fail, but it doesn't make sense to take the argument and do nothing with it.

We should either have the failing tests in the xfail list, or actually fix the problem by using this argument.

Making the test pass w/o fixing the issue makes it harder to fix the problem later.

Comment (TamaraAtanasoska) [reply fairlearn/adversarial/_adversarial_mitigation.py:544]: Oh I see what you mean, it is better to xfail it until we know if there is an actual use case for it for fairlearn. Makes sense, I'll push a commit now.

Comment (adrinjalali) [reply fairlearn/adversarial/_adversarial_mitigation.py:544]: Yes, but you still have this unused argument here. Please remove the argument, to make sure later the relevant test fails.

**Comment by TamaraAtanasoska** [comment]
@adrinjalali all checks have passed now! I removed the lightgbm verbose output as it was making it hard to scroll to see what the doc build failures are in [9fa8bbe](https://github.com/fairlearn/fairlearn/pull/1384/commits/9fa8bbe9ce30f814c6a172a04505610ed1b49e7c), in line with the discussion in [this PR](https://github.com/Microsoft/LightGBM/issues/1157). It is a bonus at the end, few changes.

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-numpy>=1.24.4
+numpy>=1.24.4, <2.0.0
```
Comment (adrinjalali) [inline requirements.txt:1]: remove this?

**Review by adrinjalali — COMMENTED** [review]
Otherwise LGTM.

**Comment by adrinjalali** [comment]
I don't get the circle CI issue, rerunning the CI.

**Comment by TamaraAtanasoska** [comment]
> I don't get the circle CI issue, rerunning the CI.

Ah interesting! I just merged the other branch in here, I will let you know when all checks pass.

**Comment by TamaraAtanasoska** [comment]
@adrinjalali all checks have now passed

**Review by adrinjalali — APPROVED** [review]
Nice, now you can get to fixing those skipped tests one by one.

**Pull request merged**